### PR TITLE
added job model, jobs routes, and accompanying tests

### PR DESCRIPTION
**Pull request by cwverica**
Everything works, except not properly throwing an error when not found.

### CODE REVIEW

File: models/job.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                    title,
+                    salary,
+                    equity,
+                    company_handle AS "companyHandle"
+           FROM jobs
+           ORDER BY title`);
+
+        let jobs = jobsRes.rows;
+
+        if (filters) {
```
Comment (montrealist) [inline models/job.js]: Could refactor this into a general `findAll` and a helper function that does the optional filtering.

File: models/job.test.js [hunk 2]
```diff
@@ -0,0 +1,255 @@
+const db = require("../db.js");
```
Comment (montrealist) [inline models/job.test.js:1]: Nice way to separate the DB concern out into its own file!

File: models/job.test.js [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        });
+
+        const result = await db.query(
+            `SELECT id, title, salary, equity, company_handle AS "companyHandle"
+             FROM jobs
+             WHERE title = 'NJ1'`);
+        expect(result.rows).toEqual([
+            {
+                id: expect.any(Number),
+                title: "NJ1",
```
Comment (montrealist) [inline models/job.test.js]: Could probably use the same `newJob` object here as well, to minimize risk of errors due to copy-paste.

File: routes/_testCommon.js [hunk 4]
```diff
@@ -3,15 +3,19 @@
 const db = require("../db.js");
 const User = require("../models/user");
 const Company = require("../models/company");
+const Job = require("../models/job");
 const { createToken } = require("../helpers/tokens");
-const { create } = require("../models/company");
+const { JsonWebTokenError } = require("jsonwebtoken");
```
Comment (montrealist) [inline routes/_testCommon.js]: Is this being used anywhere?

**Review by montrealist — APPROVED** [review]
Looks good!

File: routes/_testCommon.js (hunk 4 above)
Comment (cwverica) [reply routes/_testCommon.js]: Nope, had it for a reason, realized I didn't need it there. Will remove.

File: models/job.js (hunk 1 above)
Comment (cwverica) [reply models/job.js]: Did this. Works for the job model filtering and the company model filtering.

**Pull request merged**